Backend/lib/beckn-spec/src/BecknV2/OnDemand/Tags.hs

```diff
@@ -220,3 +252,53 @@ data Tag
     RIDER_PHONE_NUMBER
   | SHOULD_FAVOURITE_DRIVER
   deriving (Show, Eq, Generic, ToJSON, FromJSON)
+
+instance Spec.CompleteTag BecknTag where
+  type TagGroupF BecknTag = BecknTagGroup
+
+  getTagDisplay = \case
+    _ -> False -- All tags are False by default, Textdd specific tags here that you want to display
+
+  -- getDescriptor :: tags -> (description, shortDescription)
+  getTagDescriptor tag = uncurry (Spec.Descriptor . Just . T.pack $ show tag) $ case tag of
+    DISTANCE_INFO_IN_M -> (Just "Distance Information In Meters", Nothing)
+    DURATION_INFO_IN_S -> (Just "Duration Information In Seconds", Nothing)
+    RETURN_TIME -> (Just "Return time in UTC", Nothing)
+    ROUND_TRIP -> (Just "Round trip", Nothing)
+    WAYPOINTS -> (Just "WAYPOINTS", Nothing)
+    MULTIPLE_ROUTES -> (Just "Multiple Routes", Nothing)
+    _ -> (Just $ convertToSentence tag, Nothing) -- TODO: move all the tags to this function and remove (_ -> case statement)
+
+  getFullTag tag = Spec.Tag (Just $ Spec.getTagDescriptor tag) (Just $ Spec.getTagDisplay tag)
+
+  getTagGroup = \case
+    DISTANCE_INFO_IN_M -> ROUTE_INFO
+    DURATION_INFO_IN_S -> ROUTE_INFO
+    RETURN_TIME -> ROUTE_INFO
+    ROUND_TRIP -> ROUTE_INFO
+    WAYPOINTS -> ROUTE_INFO
+    MULTIPLE_ROUTES -> ROUTE_INFO
+    a -> error $ "getTagGroup function of CompleteTag class is not defined for " <> T.pack (show a) <> " tag" -- TODO: add all here dheemey dheemey (looks risky but can be catched in review and testing of feature, will be removed once all are moved to this)
+
+convertToSentence :: Show a => a -> Text
+convertToSentence = T.pack . toSentence . show
+  where
+    toSentence [] = []
+    toSentence (x : xs) = toUpper x : map toLower' xs
+
+    toLower' '_' = ' '
+    toLower' c = toLower c
+
+convertToTagGroup :: TagList -> [Spec.TagGroup]
+convertToTagGroup tagList = do
```

Comment:
The return type should be `Maybe [Spec.TagGroup]`

In case of empty array, we shouldn't send the field at all.